testsuite/src/main/java/io/netty/testsuite/transport/AbstractSingleThreadEventLoopTest.java

```diff
@@ -35,19 +50,108 @@ public void testChannelsRegistered()  {
             final Channel ch1 = newChannel();
             final Channel ch2 = newChannel();
 
-            assertEquals(0, loop.registeredChannels());
+            int rc = loop.registeredChannels();
+            boolean channelCountSupported = rc != -1;
+
+            if (channelCountSupported) {
+                assertEquals(0, loop.registeredChannels());
+            }
 
             assertTrue(loop.register(ch1).syncUninterruptibly().isSuccess());
             assertTrue(loop.register(ch2).syncUninterruptibly().isSuccess());
-            assertEquals(2, loop.registeredChannels());
+            if (channelCountSupported) {
+                assertEquals(2, loop.registeredChannels());
+            }
 
             assertTrue(ch1.deregister().syncUninterruptibly().isSuccess());
-            assertEquals(1, loop.registeredChannels());
+            if (channelCountSupported) {
+                assertEquals(1, loop.registeredChannels());
+            }
         } finally {
             group.shutdownGracefully();
         }
     }
 
+    @Test
+    @SuppressWarnings("deprecation")
+    public void shutdownBeforeStart() throws Exception {
+        EventLoopGroup group = newEventLoopGroup();
+        assertFalse(group.awaitTermination(2, TimeUnit.MILLISECONDS));
+        group.shutdown();
+        assertTrue(group.awaitTermination(200, TimeUnit.MILLISECONDS));
+    }
+
+    @Test
+    public void shutdownGracefullyZeroQuietBeforeStart() throws Exception {
+        EventLoopGroup group = newEventLoopGroup();
+        assertTrue(group.shutdownGracefully(0L, 2L, TimeUnit.SECONDS).await(200L));
```

Comment:
200ms seems a bit low, I see this failing on a loaded CI server.

Reply:
Maybe it could be bumped a little, but 200ms seems to me like it should be long enough. The timeouts are intentionally low (< 1 sec) since that's what's currently needed to expose the underlying problem. As explained in the description the bug is largely "masked" by the 1sec default wait/select timeout.